### PR TITLE
New icons?

### PR DESCRIPTION
**Pull request by tonsky**
I love using clj-async-profiler, but one thing I noticed is that icons are confusing to me. ↙↗ reads like “fullscreen” and 🔄 reads like “reload”. I drew new ones, but if you don’t like them we can discuss how you’d want them to look. Let me know!

Screenshot 2024-11-19 at 15 16 31
Screenshot 2024-11-19 at 15 16 35

### CODE REVIEW

**Comment by alexander-yakushev** [comment]
Thank you for tackling this! The current icons are just some random stuff from the web. I've tried to make them match one another, but I don't have a strong attachment to them. Your icons are 100% nicer to look at but I'd like to clarify some points about the UX part.

The current arrows for collapse/expand sidebar don't bother me much. Being diagonal they also have a nice property of hinting that the action will change the size both horizontally and vertically which is indeed what happens. It's by no means obligatory; just a small pleasing detail. What I dislike about them is that they are too "anemic" when compared to other icons. Your icons, on the other hand, are better because they have a clear outline that matches the others well. I also really appreciate that your icons have arrows too. For example, I to this day can't quckly recognize what Github's "Collapse sidebar" button does, like, what the hell:
image
Is there a reason why expand has a vertical divider and collapse does not? I see that Github's interface has similar buttons for collapse/expand and they both have the dividers.

For your reversal icons, I really _really_ like how they look. It is incredibly witty way to put this very recognizable state of the flamegraph into a minimalistic icon. The only thing I'm afraid of is that while they represent the current state perfectly, as a button they don't compel me as a user to click them because the first time I'm not sure what happens when I do. Sure, the button is accompanied by both a label and a tooltip, so it's not a big deal. And you are right that the current icon can be confused with "refresh" so it doesn't accurately tell the user what happens either.

File: res/flamegraph-rendering/template.html [hunk 1]
```diff
@@ -48,15 +48,11 @@
...
       .btn:active {transform:scale(0.90);}
       .toggle {background:none; border:none; padding:6px; transition: all 0.3s ease;}
-      .toggle svg {width:16px; height:16px; fill:#006d5b; stroke:#006d5b;
-                   transition: all 0.3s ease;}
-      .toggle.arcs svg {fill:white;}
+      .toggle svg {width:16px; height:16px; fill: white; stroke:#006d5b; transition: all 0.3s ease;}
       .toggle:hover {background-color: #e6f2ff; border-radius: 5px;}
       .toggle.arcs:hover svg {fill:#e6f2ff;}
       .toggle:active {transform:scale(0.90);}
-      .toggle.toggled {background-color: #006d5b; border-radius: 5px;}
```
Comment (alexander-yakushev) [inline res/flamegraph-rendering/template.html:57]: This "pressed in" effect is also used for "Highlight" – do you want to eliminate it there too?

**Comment by tonsky** [comment]
> Is there a reason why expand has a vertical divider and collapse does not?

The divider illustrates the sidebar. Expand shows that it will appear, collapse shows that it will go away. But maybe I was overthinking it, I can put it on both, so they will only differ by arrow direction.

> the first time I'm not sure what happens when I do

That’s fine. Some actions have no well-established icon, so we’ll have to go with one which meaning you’ll have to learn. In that case what’s important is that you’ll remember what it does and, after learning, find it easily.

Using refresh icon here is strictly worse because it does have well-established meaning and as a result confuses users.

**Comment by tonsky** [comment]
Okay I restored highlight on search icon, made collapse icon match expand, and updated paddings on smallbar. WDYT?

https://github.com/user-attachments/assets/bb63794e-4e2a-4dea-a7d6-5f793b6c984e

**Comment by alexander-yakushev** [comment]
> But maybe I was overthinking it, I can put it on both, so they will only differ by arrow direction.

Thank you, I think that Collapse looks better with the divider. Without a divider, it seems to be a button that is going to open a context menu or something like that.

> Okay I restored highlight on search icon, made collapse icon match expand, and updated paddings on smallbar. WDYT?

1. Reverse icon looks awesome, and the flipping animation is really cute.
2. Highlighting whole background instead of outline on hover (and when toggled too) looks good, I like it.
3. I intentionally aligned sidebar and smallbar and the icons in them in such a way that the three icons on the smallbar stay at the exact absolute locations on collapse/expand. Same goes for the top and right borders of the smallbar. I want to retain this quality. This is why you see that extra top padding on the smallbar (and bottom padding for symmetry). If it can be done differently, I'm open to suggestions.

**Comment by tonsky** [comment]
Right! I restored the padding. I agree small bar and expanded should be addressed together, maybe in another PR later

**Comment by alexander-yakushev** [comment]
Excellent, thank you! I can clearly see now why you wanted to reduce the padding – buttons with the new icons definitely feel a bit too small, probably because the new icons contain more small details. But we can address this later.

**Pull request merged**